qiskit_rigetti/_qcs_provider.py

```diff
@@ -82,7 +82,7 @@ def backends(self, name: Optional[str] = None, **__: Any) -> List[RigettiQCSBack
             return self._backends
         return [b for b in self._backends if b.name() == name]
 
-    def get_simulator(self, *, num_qubits: int, noisy: bool = False) -> RigettiQCSBackend:
+    def get_simulator(self, *, name: str = "", num_qubits: int = None, noisy: bool = False) -> RigettiQCSBackend:
```

Comment:
It's been a while since I used Qiskit, this constructor is usually called directly by users, right? Not called internally by Qiskit?

The reason I ask is I think it'd be better to require _either_ name _or_ num_qubits so that folks don't accidentally wind up with something like `Noneq-qvm` and a weird error message. One way we could do this is with two separate methods, `get_simulator` as-is and `get_simulator_by_name` with the new arg. Alternatively, if they need to be the same method, maybe we can use [`@overload`](https://docs.python.org/3/library/typing.html?highlight=overload#typing.overload) to signal the intent?

Reply:
Yeah it's called by the users.

I suggest we wrap it more closely around the pyquil `get_qc` which handles these edge cases anyway. This isn't checked on construction of the backend but only when you call `run`. To me it would make sense to do the check on construction for faster feedback of if you have given it something like `Noneq-qvm`

Reply:
Having a more consistent experience with pyquil sounds like generally a good thing—I assumed we had split out `get_simulator` to follow a common Qiskit pattern. Maybe we can just have `get_backend` work like pyquil's `get_qc`? Then folks could pass arbitrary qvm strings in there for more advanced usecases.